sources/api/shibaken/src/warmpool/autoscaling_warm_pool.rs

```diff
@@ -0,0 +1,129 @@
+/// This module contains utilities for waiting until the instance on warm pool to be
+/// marked as InService.
+use crate::warmpool::error::{self, Result};
+use argh::FromArgs;
+use imdsclient::ImdsClient;
+use serde::Deserialize;
+use snafu::{OptionExt, ResultExt};
+use std::fs;
+use std::path::Path;
+use tokio::time::{sleep, Duration};
+
+// Path to config file containing host's autoscaling.should-wait setting
+const CONFIG_PATH: &str = "/etc/warm-pool-wait.toml";
+const FETCH_LIFECYCLE_INTERVAL_SECS: Duration = Duration::from_secs(5);
+
+#[derive(FromArgs, Debug)]
+#[argh(subcommand, name = "warm-pool-wait")]
+/// Wait until the instance reaches the `InService` state
+pub(crate) struct WarmPoolWait {}
+
+impl WarmPoolWait {
+    pub(crate) async fn run(self) -> Result<()> {
+        let config_parse = get_config().await?;
+
+        let should_wait_value = config_parse.should_wait;
+        let marker_file_path = config_parse.marker_path;
+
+        println!("autoscaling.should-wait value is {}", should_wait_value);
+        if should_wait_value {
+            wait_until_inservice().await?;
+        }
+
+        println!("Marker file path is {}", marker_file_path);
+        fs::write(&marker_file_path, "").unwrap_or_else(|e| {
+            println!("Failed to create marker file '{}', warm-pool-wait service may unexpectedly run again: '{}'",
+            &marker_file_path, e);
+        });
+        Ok(())
+    }
+}
+
+/// Uses an imdsclient function to fetch the lifecyclestate of the host
+async fn get_lifecycle_state(client: &mut ImdsClient) -> Result<String> {
+    client
+        .fetch_autoscaling_lifecycle_state()
+        .await
+        .context(error::ImdsRequestSnafu)?
+        .context(error::ImdsNoneSnafu {
+            what: "target-lifecycle-state",
+        })
+}
+
+/// Continuously fetches lifecycle state of the host.
+/// Returns from function when host is "InService"
+async fn wait_until_inservice() -> Result<()> {
+    let mut client = ImdsClient::new();
+
+    loop {
+        let lifecycle_state = get_lifecycle_state(&mut client).await?;
+
+        if lifecycle_state.eq("InService") {
+            println!("Lifecycle state is {} ... exiting.", lifecycle_state);
+            break;
+        } else {
+            println!("Lifecycle state is {} ... waiting.", lifecycle_state);
```

Comment:
Couldn't this loop go on for a long time? Do we want to print so many of these? Should we print one every minute instead of 20 times per minute?

Reply:
This loop goes on for a very short time.